docs/source/opentelemetry.mdx

```diff
@@ -0,0 +1,214 @@
+
+[OpenTelemetry](https://opentelemetry.io/) is a collection of open-source tools for generating and processing telemetry data (such as logs and metrics) from different systems in a generic, consistent way.
+
+You can configure your gateway, your individual subgraphs, or even a monolothic Apollo Server instance to emit telemetry related to processing GraphQL operations.
+
+Additionally, the `@apollo/gateway` library provides built-in OpenTelemetry instrumentation to emit [gateway-specific spans](#gateway-specific-spans) for operation traces.
+
+> Apollo Studio does _not_ currently consume OpenTelemetry-formatted data. To push trace data to Studio, see [Federated trace data](./metrics/).
+>
+> You should configure OpenTelemetry if you want to push trace data to an OpenTelemetry-compatible system, such as [Zipkin](https://zipkin.io/) or [Jaeger](https://www.jaegertracing.io/).
+
+## Setup
+
+### 1. Install required libraries
+
+To use OpenTelemetry in your application, we recommend installing _at least_ the following `@opentelemetry` Node.js libraries (you can install additional instrumentation libraries as desired):
+
+```bash
+npm install \
+  @opentelemetry/api \
+  @opentelemetry/node@0.22 \
+  @opentelemetry/core@0.22 \
+  @opentelemetry/instrumentation-http@0.22 \
+  @opentelemetry/instrumentation-express@0.22 \
+  @opentelemetry/instrumentation-graphql@0.22
+```
+
+> As shown, most `@opentelemetry` libraries should maintain a consistent version number to prevent dependency conflicts. As of this article's most recent update, the latest version is `0.22`. 
+
+### 2. Configure instrumentation
+
+Next, update your application to configure your OpenTelemetry instrumentation _as early as possible in your app's execution_. This _must_ occur before you even import `apollo-server`, `express`, or `http`. Otherwise, your trace data will be incomplete.
+
+**We recommend putting this configuration in its own file**, which you import at the very top of `index.js`.
+
+> Note that for now, this code does _not_ push trace data to an external system. Instead, it prints that data to the console for debugging purposes.
+
+```js:title=open-telemetry.js
+// Import required symbols
+const { HttpInstrumentation } = require ('@opentelemetry/instrumentation-http');
+const { ExpressInstrumentation } = require ('@opentelemetry/instrumentation-express');
+const { GraphQLInstrumentation } = require ('@opentelemetry/instrumentation-graphql');
+const { registerInstrumentations } = require('@opentelemetry/instrumentation');
+const { NodeTracerProvider } = require("@opentelemetry/node");
+const { SimpleSpanProcessor, ConsoleSpanExporter } = require ("@opentelemetry/tracing");
+const { Resource } = require('@opentelemetry/resources');
+
+// Register server-related instrumentation
+registerInstrumentations({
+  instrumentations: [
+    new HttpInstrumentation(),
+    new ExpressInstrumentation(),
+    new GraphQLInstrumentation()
```

Comment:
`new GraphQLInstrumentation()` must not be included for Federation as it breaks stuff.
It must be included for ApolloServer though.